src/main/java/namor/template/calculator/RPNcalculator.java

```diff
@@ -1,67 +1,71 @@
 package namor.template.calculator;
 
-import namor.template.stack.ArrayStack;
+import namor.template.stack.Stack;
 
 public class RPNcalculator {
 
+    private Stack<Double> operands;
 
-    public int calculate(String arithmeticExpression) {
-        ArrayStack numbers = new ArrayStack();
+    public RPNcalculator(Stack operands) {
+        this.operands = operands;
+    }
+
+    public double calculate(String arithmeticExpression) {
 
         for (String operationElement : arithmeticExpression.split(" ")) {
             if (Sign.isSign(operationElement)) {
-                doArithmeticOperation(operationElement, numbers);
+                doArithmeticOperation(operationElement, operands);
             } else {
-                numbers.push(Integer.parseInt(operationElement));
+                operands.push(Double.parseDouble(operationElement));
             }
         }
-        return numbers.pop();
+        return operands.pop();
     }
 
-    private void doArithmeticOperation(String arithmeticOperator, ArrayStack numbers) {
+    private void doArithmeticOperation(String arithmeticOperator, Stack numbers) {
         switch (Sign.getValue(arithmeticOperator)) {
             case ADDITION:
-                numbers.push(addition(numbers));
+                numbers.push(add(numbers));
                 break;
             case SUBTRACTION:
-                numbers.push(subtraction(numbers));
+                numbers.push(subtract(numbers));
                 break;
             case MULTIPLICATION:
                 numbers.push(multiplication(numbers));
                 break;
             case DIVISION:
-                numbers.push(division(numbers));
+                numbers.push(divide(numbers));
```

Comment:
I recommend to avoid make method calls inside of another method like 'instance.method1(method2())'. Two reasons first it's less readable. second - it harder to debug. 

you can make some variable 'double result'. And in switch assign operation result to that variable. and after switch statement push into stack.